apis/refs/v1beta1/sqldatabaseref.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type SQLDatabaseRef struct {
+	/* The SQL Database selfLink, when not managed by Config Connector. */
+	External string `json:"external,omitempty"`
+	/* The `name` field of a `SQLDatabase` resource. */
+	Name string `json:"name,omitempty"`
+	/* The `namespace` field of a `SQLDatabase` resource. */
+	Namespace string `json:"namespace,omitempty"`
+}
+
+type SQLDatabase struct {
+	ProjectID  string
+	InstanceID string
+	DatabaseID string
+}
+
+func (s *SQLDatabase) String() string {
+	return "projects/" + s.ProjectID + "/instances/" + s.InstanceID + "/databases/" + s.DatabaseID
+}
+
+func ResolveSQLDatabaseRef(ctx context.Context, reader client.Reader, obj client.Object, ref *SQLDatabaseRef) (*SQLDatabase, error) {
+	if ref == nil {
+		return nil, nil
+	}
+
+	if ref.Name == "" && ref.External == "" {
+		return nil, fmt.Errorf("must specify either name or external on databaseRef")
+	}
+	if ref.External != "" && ref.Name != "" {
+		return nil, fmt.Errorf("cannot specify both spec.databaseRef.name and spec.databaseRef.external")
+	}
+
+	if ref.External != "" {
```

Comment:
Users should be able to provide database name directly via the `external` field. e.g.

```
databaseRef:
  external: sqldatabase-sample-${uniqueId}
```